roles/common/tasks/host_validations.yml

```diff
@@ -78,14 +78,110 @@
   tags:
   - validate
   - validate_tmp_access
-  
-  - name: Validate enough free disk space for package installation
+
+- name: Validate enough free disk space for package installation
   assert:
     that: item.size_available > {{ required_disk_space_mb }}|float
-    fail_msg: >- 
-      "Not enough free disk space for package installation.  Minimum space required is {{ required_disk_space_mb }}MB, please free up more disk space 
+    fail_msg: >-
+      "Not enough free disk space for package installation.  Minimum space required is {{ required_disk_space_mb }}MB, please free up more disk space
       to continue installation. To skip host validations, set validate_hosts to false."
   with_items: "{{ ansible_mounts }}"
-  tags: 
+  tags:
     - validate
     - validate_disk_usage
+
+- name: Validate enough free memory for Zookeeper hosts
+  assert:
+    that: ansible_memtotal_mb > {{ required_total_memory_mb_zookeeper }}|float
+    fail_msg: >-
+      "Not enough memory to run Zookeeper.  Minimum memory required is {{ required_total_memory_mb_zookeeper }}MB, please add more memory to host
+      or choose a different host to install Zookeeper to continue installation."
+  when: groups['zookeeper'] is defined and inventory_hostname in groups['zookeeper']
+  tags:
+    - validate
+    - validate_memory_usage
+    - validate_memory_usage_zookeeper
+
+- name: Validate enough free memory for Kafka Broker hosts
```

Comment:
Can we use some loop here to reuse the code

Reply:
Can you clarify where you think a loop should be specified here?

Reply:
Nevermind, not much of benefit.